tests/sources/test_graphql.py

```diff
@@ -0,0 +1,225 @@
+        authentication_method="none",
+        graphql_query=graphql_query,
+        object_list=["users", "name"],
+        headers=headers,
+        graphql_variables=graphql_variables,
+    ) as source:
+        yield source
+
+
+@pytest.mark.asyncio
+async def test_get():
+    async with create_graphql_source() as source:
+        source.graphql_client._get_session.get = Mock(
+            return_value=get_json_mock(
+                mock_response={"data": {"users": "xyz"}}, status=200
+            )
+        )
+        data = await source.graphql_client.get("query")
+        assert data == {"users": "xyz"}
+
+
+@pytest.mark.asyncio
+@patch("connectors.utils.time_to_sleep_between_retries", Mock(return_value=0))
+async def test_get_with_errors():
+    async with create_graphql_source() as source:
+        source.graphql_client._get_session.get = Mock(
+            return_value=get_json_mock(
+                mock_response={
+                    "errors": [{"type": "QUERY", "message": "Invalid query"}]
+                },
+                status=200,
+            )
+        )
+        with pytest.raises(Exception):
+            await source.graphql_client.get("query")
+
+
+@pytest.mark.asyncio
+async def test_post():
+    async with create_graphql_source() as source:
+        source.graphql_client._get_session.post = Mock(
+            return_value=get_json_mock(
+                mock_response={"data": {"users": "xyz"}}, status=200
+            )
+        )
+        data = await source.graphql_client.post("query")
+        assert data == {"users": "xyz"}
+
+
+@pytest.mark.asyncio
+@patch("connectors.utils.time_to_sleep_between_retries", Mock(return_value=0))
+async def test_post_with_errors():
+    async with create_graphql_source() as source:
+        source.graphql_client._get_session.post = Mock(
+            return_value=get_json_mock(
+                mock_response={
+                    "errors": [{"type": "QUERY", "message": "Invalid query"}]
+                },
+                status=200,
+            )
+        )
+        with pytest.raises(Exception):
+            await source.graphql_client.post("query")
+
+
+@pytest.mark.asyncio
+@patch("connectors.utils.time_to_sleep_between_retries", Mock(return_value=0))
+async def test_make_request_with_unauthorized():
+    async with create_graphql_source() as source:
+        source.graphql_client._get_session.post = Mock(
+            side_effect=ClientResponseError(
+                status=401,
+                request_info=aiohttp.RequestInfo(
+                    real_url="", method=None, headers=None, url=""
+                ),
+                history=None,
+            )
+        )
+        with pytest.raises(UnauthorizedException):
+            await source.graphql_client.make_request("QUERY")
+
+
+@pytest.mark.asyncio
+@patch("connectors.utils.time_to_sleep_between_retries", Mock(return_value=0))
+async def test_make_request_with_429_exception():
```

Comment:
I see we've a test for `429` errors, but we don't handle it anywhere in the code explicitly right?

Reply:
I have added to test `clienterror` other that `401`. Regarding `429` we do not know the source so how we can handle the `429` as all source have different header name or API to fetch retry-after.